src/generator-create-api-mesh.js

```diff
@@ -22,32 +22,50 @@ class ApiMeshCreateGenerator extends Generator {
   }
 
   async install () {
-    try {
-      this.log('Check if Mesh plugin is installed')
-      const listInstalledPlugins = (await this.spawnCommand('aio', ['plugins'], { stdio: [process.stderr] })).stdout
-      const isPluginInstalled = listInstalledPlugins.indexOf(meshPluginName) !== -1
-      if (isPluginInstalled) {
-        this.log(meshPluginName + ' is installed')
-      } else {
-        this.log(meshPluginName + ' is not installed. Installing...')
+    this.log('Check if Mesh plugin is installed')
+    const listInstalledPlugins = (await this.spawnCommand('aio', ['plugins'], { stdio: [process.stderr] })).stdout
+    const isPluginInstalled = listInstalledPlugins.indexOf(meshPluginName) !== -1
+    if (isPluginInstalled) {
+      this.log(meshPluginName + ' is installed')
+    } else {
+      this.log(meshPluginName + ' is not installed. Installing...')
+      try {
         await this.spawnCommand('aio', ['plugins', 'install', meshPluginName])
+      } catch(e) {
+        throw new Error('Unable to create a mesh. Run "aio plugins" and check if "@adobe/aio-cli-plugin-api-mesh" plugin is installed. Make sure you\'re using the latest version.')
       }
-      const output = (await this.spawnCommand('aio', ['api-mesh', 'create', '-c', this.options['template-folder'] + '/conf/mesh.json', '--json'], { stdio: [process.stderr] })).stdout
-
-      this.props.meshConfig = JSON.parse(output.substring(output.indexOf('{'), output.lastIndexOf('}') + 1))
-      const dotenvFile = this.destinationPath(constants.dotenvFilename)
-      const vars = [{ name: 'MESH_ID', value: this.props.meshConfig.mesh.meshId }, { name: 'MESH_API_KEY', value: this.props.meshConfig.adobeIdIntegrationsForWorkspace.apiKey }]
-      const content = `${vars.map(v => `${v.name}=${v.value}`).join(EOL)}${EOL}`
-      /* istanbul ignore next */
-      if (this.fs.exists(dotenvFile)) {
+    }
+
+    this.log('checking if selected workspace doesn\'t have a mesh');
+
+    let isMeshFound = false;
+    try {
+       await this.spawnCommand('aio', ['api-mesh', 'get','--json'], { stdio: [process.stderr] });
+       isMeshFound = true;
+    } catch(err) {
```

Comment:
Is there a specific error type that is being caught here? Maybe something we can use to determine that the error is related to the mesh not being found? My guess is no, but just double-checking.

Reply:
Yes, @amolina-adobe I inspected the err object as well as the output, we don't get any status codes / other indicators to identify mesh not being found. I found  only the stderr returns a message. So I used it verify mesh status.